### ISSUE TITLE
Automatic testing suite

### ISSUE THREAD

**Issue by bmackinney** [pantheon-systems/documentation#418]
connects to #874
Tests: 
- [homepage feature](https://github.com/pantheon-systems/documentation/blob/master/features/homepage.feature)

Tests to add:
- [ ] linkchecker
- Behat
  - [x] CTA
  - [x] Edit button
  - [x] Search
  - [x] [Residual text from merge conflicts](https://github.com/pantheon-systems/documentation/issues/418#issuecomment-139362177)
- [ ] Visual diffs?
- [ ] Ghost inspector?
- [ ] cross-browser tests
- [ ] write-good linter

**Comment by rachelwhitton** [comment]
Moving Behat into this issue. 

Here's what I have so far: 

Current Scenarios: 
-  Scenario: The homepage text includes expected content, such as links to all articles, etc.
-  Scenario: Verify Contributor CTA button and links exists

I pushed branch [874-behat-tests](https://github.com/pantheon-systems/documentation/tree/874-behat-tests) which includes a javascript scenario. Currently uses default browser (FireFox) of driver. Using FF because for some reason a few feature context methods aren't working in Chrome (e.g. press and click for testing contributor button). I can add more info about replicating the Chrome issues if you're interested. Otherwise, I'll keep using FireFox. 

<hr>
Here's more about the problem with Chrome - I'll check versions tomorrow for Selenium.

behat version 3.0.15

Adding `browser: chrome` to `selenium2:` in `behat.yml` causes test to fail:
```
    And I press "Contribute to Docs"  # FeatureContext::pressButton()
      Button with id|name|title|alt|value "Contribute to Docs" not found.

2 scenarios (1 passed, 1 failed)
18 steps (14 passed, 3 skipped, 1 failed)
```

I wrote up a [FeatureContext method](https://github.com/pantheon-systems/documentation/blob/874-behat-tests/features/bootstrap/FeatureContext.php#L22-L37) to check if a provided element ID exists. Replaced the above step with `Then I find the element with ID "cta-contrib-btn`and re-run the test. This time, the element is found and step is passed, but remaining tests fail because the element was not toggled onclick. 

I tried to toggle the dropdown button by adding `$elemendID->click();` to the [FeatureContext method](https://github.com/pantheon-systems/documentation/blob/874-behat-tests/features/bootstrap/FeatureContext.php#L22-L37) but that gives the following fatal error:
```
PHP Fatal error:  Call to a member function click() on a non-object in /Users/Rachelwhitton/sites/documentation/features/bootstrap/FeatureContext.php on line 36
```

**Comment by rachelwhitton** [comment]
@bmackinney - can we include a test to check all pages for residual merge conflicts? <hr>
![screen shot 2015-09-10 at 2 46 04 pm](https://cloud.githubusercontent.com/assets/10119525/9799080/b3057046-57ca-11e5-8848-da7513149d35.png)

These can be identified by searching all pages for `<<<<<<< HEAD`, `=======`, and `>>>>>>>` text

I'd need to work on finding or creating logic to search all pages.

**Comment by bmackinney** [comment]
Yes.

**Comment by rachelwhitton** [comment]
@brian @ari-gold I was able to debug the issue with chrome by updating a few dependencies to our composer.json file, while I was at it I went ahead and ran `composer validate` and resolved all errors/notices concerning unbound version constraints and implemented composer's recommendations.

Check out the diff and let me know if you have any feedback and/or concerns: [874-behat-tests](https://github.com/pantheon-systems/documentation/compare/418-automatic-testing-suite)

PR will come soon hopefully for at least behat tests. @bmackinney whats an acceptable MVP to ship for this suite? I suggest we split this issue up by MVP and have them under the Automation milestone.

**Comment by greg-1-anderson** [comment]
One of the interesting things about CI with Drupal 8 is that it is not convenient to import your site's configuration on a Database that is different than the one it was exported from.  See  drush-ops/drush#1625 and drush-ops/drush#1372.  This is an issue for CI, because in CI test, usually an empty site is installed rather than pulling a database from live.  It is harder to get config imported into a new empty site, though. Solutions still in progress in previously-mentioned issues.

**Comment by bmackinney** [comment]
closed by #1134
